lighthouse-core/audits/unused-css-rules.js

```diff
@@ -0,0 +1,154 @@
+      requiredArtifacts: ['CSSUsage', 'Styles']
+    };
+  }
+
+  /**
+   * @param {!Array.<Object>} styles The output of the Styles gatherer.
+   * @return {!Object} A map of styleSheetId to stylesheet information.
+   */
+  static indexStylesheetsById(styles) {
+    return styles.reduce((indexed, stylesheet) => {
+      indexed[stylesheet.header.styleSheetId] = Object.assign({
+        used: [],
+        unused: [],
+      }, stylesheet);
+      return indexed;
+    }, {});
+  }
+
+  /**
+   * Counts the number of unused rules and adds count information to sheets.
+   * @param {!Array.<RuleUsage>} rules The output of the CSSUsage gatherer.
+   * @param {!Object} indexedStylesheets Stylesheet information indexed by id.
+   * @return {number} The number of unused rules.
+   */
+  static countUnusedRules(rules, indexedStylesheets) {
+    let unused = 0;
+
+    rules.forEach(rule => {
+      const stylesheetInfo = indexedStylesheets[rule.styleSheetId];
+
+      if (rule.used) {
+        stylesheetInfo.used.push(rule);
+      } else {
+        unused++;
+        stylesheetInfo.unused.push(rule);
+      }
+    });
+
+    return unused;
+  }
+
+  /**
+   * @param {Object} stylesheetInfo The stylesheetInfo object.
+   * @return {!{url: string, label: string, code: string}} The result for the URLLIST formatter.
+   */
+  static mapSheetToResult(stylesheetInfo) {
+    const numUsed = stylesheetInfo.used.length;
+    const numUnused = stylesheetInfo.unused.length;
+    const percentUsed = Math.round(100 * numUsed / (numUsed + numUnused)) || 0;
+
+    let contentPreview = stylesheetInfo.content;
+    if (contentPreview.length > PREVIEW_LENGTH) {
+      const firstRuleStart = contentPreview.indexOf('{');
+      const firstRuleEnd = contentPreview.indexOf('}');
+      if (firstRuleStart === -1 || firstRuleEnd === -1
+          || firstRuleStart > firstRuleEnd
+          || firstRuleStart > PREVIEW_LENGTH) {
+        contentPreview = contentPreview.slice(0, PREVIEW_LENGTH) + '...';
+      } else if (firstRuleEnd < PREVIEW_LENGTH) {
+        contentPreview = contentPreview.slice(0, firstRuleEnd + 1) + ' ...';
+      } else {
+        const lastSemicolonIndex = contentPreview.slice(0, PREVIEW_LENGTH).lastIndexOf(';');
+        contentPreview = lastSemicolonIndex < firstRuleStart ?
+            contentPreview.slice(0, PREVIEW_LENGTH) + '... } ...' :
+            contentPreview.slice(0, lastSemicolonIndex + 1) + ' ... } ...';
+      }
+    }
+
+    return {
+      url: stylesheetInfo.header.sourceURL || 'inline',
+      label: `${percentUsed}% rules used`,
+      code: contentPreview.trim(),
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const styles = artifacts.Styles;
+    const usage = artifacts.CSSUsage;
+
+    if (typeof styles === 'undefined' || typeof usage === 'undefined') {
```

Comment:
this first block is no longer necessary because [runner now checks](https://github.com/GoogleChrome/lighthouse/blob/f98b33de666efd8330c0aef602ff7244ffe7b857/lighthouse-core/runner.js#L165-L180) that all required artifacts are defined before running an audit